File: go/FirstUniqueOccur/FirstUniqueOccur.go
package main

import "fmt"

// Time -> 0(N) Space -> 0(N)
func firstUniqChar(s string) int {
	occur := make(map[byte]int, 0)

	for char := range s {
		occur[s[char]]++
	}

	for char := range s {
		if occur[s[char]] == 1 {
			return char
		}
	}

	return -1
}

func main() {
	fmt.Println(firstUniqChar("loveleetcode"))
}

File: python/firstuniqueoccur.py
# Time -> 0(N) Space -> 0(N)
def firstUniqChar(s: str) -> int:
    occur = {}
    for i in range(len(s)):
        occur[s[i]] = occur.get(s[i], 0) + 1
    for i in range(len(s)):
        if occur[s[i]] == 1:
            return i
    return -1


print(firstUniqChar("leetcode"))
print(firstUniqChar("loveleetcode"))
